nemo/collections/nlp/models/nlp_model.py

```diff
@@ -355,6 +378,26 @@ def load_from_checkpoint(
                 model = ptl_load_state(cls, checkpoint, strict=strict, cfg=cfg, **kwargs)
                 # cfg = checkpoint[cls.CHECKPOINT_HYPER_PARAMS_KEY].cfg
 
+            # if the checkpoint is distributed, we deferred loading the state_dict until now
+            if checkpoint_dir is not None:
+                sharded_state_dict = model.sharded_state_dict()
+                checkpoint['state_dict'] = sharded_state_dict
+                # dist checkpointing needs torch.distributed to load the checkpoint
+                if parallel_state.is_unitialized():
+
+                    def dummy():
+                        return
+
+                    if model.trainer.strategy.launcher is not None:
+                        model.trainer.strategy.launcher.launch(dummy, trainer=model.trainer)
+                    model.trainer.strategy.setup_environment()
+                # load the checkpoint from disk
+                checkpoint = dist_checkpointing.load(sharded_state_dict=checkpoint, checkpoint_dir=checkpoint_dir)
```

Comment:
question: does dist_checkpointing internally check if state_dict matching is strict?